Examples/Swift/CustomViewController.swift

```diff
@@ -0,0 +1,195 @@
+        }
+
+        speak(text)
+    }
+
+    // Notifications sent on all location updates
+    func progressDidChange(_ notification: NSNotification) {
+        let routeProgress = notification.userInfo![RouteControllerAlertLevelDidChangeNotificationRouteProgressKey] as! RouteProgress
+        updateRouteProgress(routeProgress: routeProgress)
+    }
+
+    // Updates the turn banner with information about the next turn
+    func updateRouteProgress(routeProgress: RouteProgress) {
+        guard let step = routeProgress.currentLegProgress.upComingStep else { return }
+
+        if let direction = step.maneuverDirection {
+            switch direction {
+            case .slightRight:
+            self.arrowView.text = "↗️"
+            case .sharpRight, .right:
+            self.arrowView.text = "➡️"
+            case .slightLeft:
+            self.arrowView.text = "↖️"
+            case .sharpLeft, .left:
+            self.arrowView.text = "⬅️"
+            case .uTurn:
+            self.arrowView.text = "⤵️"
+            default:
+            self.arrowView.text = "⬆️"
+            }
+        }
+        self.instructionLabel.text = step.names?.first
+        let distance = routeProgress.currentLegProgress.currentStepProgress.distanceRemaining
+        self.distanceLabel.text = lengthFormatter.string(fromMeters: distance)
+    }
+
+    // Fired when the user is no longer on the route.
+    // A new route should be fetched at this time.
+    func rerouted(_ notification: NSNotification) {
+        speechSynth.stopSpeaking(at: .word)
+
+        getRoute {
+            /*
+             **IMPORTANT**
+
+             When rerouting, you need to give the RouteController a new route.
+             Otherwise, it will continue to compare the user to the old route and continually reroute the user.
+             */
+            let routeProgress = RouteProgress(route: self.userRoute!)
+            self.routeController.routeProgress = routeProgress
+            self.updateRouteProgress(routeProgress: routeProgress)
+        }
+    }
+
+    func getRoute(completion: (()->())? = nil) {
+        let options = RouteOptions(coordinates: [mapView.userLocation!.coordinate, destination.coordinate])
+        options.includesSteps = true
+        options.routeShapeResolution = .full
+        options.profileIdentifier = .automobileAvoidingTraffic
+
+        _ = directions.calculate(options) { [weak self] (waypoints, routes, error) in
+            guard error == nil else {
+                print(error!)
+                return
+            }
+
+            guard let route = routes?.first else {
+                return
+            }
+
+            self?.userRoute = route
+
+            completion?()
+            self?.addRouteToMap()
+        }
+    }
+
+    func addRouteToMap() {
+        guard let style = mapView.style else { return }
+        guard let userRoute = userRoute else { return }
+
+        mapView.addAnnotation(destination)
```

Comment:
We'll also want to clear the map of all annotations before adding a new destination.